script.domoticz.scenes/resources/lib/script.py

```diff
@@ -0,0 +1,307 @@
+    else:
+        protocol = 'http://'
+
+    url = protocol + host + ':' + str(port) + '/json.htm?' + setcreds + 'type=command&param=getversion'
+    try:
+        result = requests.get(url, verify=verifySsl)
+    except:
+        return -1
+
+    answer = result.content
+    jsonResult = json.loads(answer)
+    return(jsonResult)
+
+
+def get_all_switches(host, port, useSsl, username=None, password=None):
+    if username is not None:
+        setcreds = 'username=' + username + '&password=' + password + '&'
+    else:
+        setcreds = ''
+
+    if useSsl == True:
+        protocol = 'https://'
+    else:
+        protocol = 'http://'
+    url = protocol + host + ':' + str(port) + '/json.htm?' + setcreds + 'type=devices&filter=light&used=true&order=Name'
+
+    try:
+        result = requests.get(url)
+    except:
+        return -1
+
+    answer = result.content
+
+    jsonResult = json.loads(answer)
+    deviceResult = jsonResult['result']
+    return(deviceResult)
+
+
+def switch_scene(host, port, idx, useSsl=False, username=None, password=None):
+    if useSsl is True:
+        base_url = "https://" + host + ":" + str(port)
+    else:
+        base_url = "http://" + host + ":" + str(port)
+
+    if username is not None:
+        setcreds = 'username=' + username + '&password=' + password + '&'
+    else:
+        setcreds = ''
+
+    url = base_url + "/json.htm?" + setcreds + "type=command&param=switchscene&idx=" + str(idx) + "&switchcmd=On"
+    f = open(r"c:\tmp\debug.txt", "a")
+    f.write(url + "\n")
+    f.close()
+    requests.get(url=url)
+
+
+def switch_switch(host, port, idx, useSsl=False, username=None, password=None):
+    if useSsl is True:
+        base_url = "https://" + host + ":" + str(port)
+    else:
+        base_url = "http://" + host + ":" + str(port)
+
+    if username is not None:
+        setcreds = 'username=' + username + '&password=' + password + '&'
+    else:
+        setcreds = ''
+
+    url = base_url + "/json.htm?" + setcreds + "type=command&param=switchlight&idx=" + str(idx) + "&switchcmd=Toggle"
+    f = open(r"c:\tmp\debug.txt", "a")
+    f.write(url + "\n")
+    f.close()
+    requests.get(url=url)
+
+
+def switch_dimmer(host, port, idx, state, useSsl=False, username=None, password=None):
+    if useSsl is True:
+        base_url = "https://" + host + ":" + str(port)
+    else:
+        base_url = "http://" + host + ":" + str(port)
+
+    if username is not None:
+        setcreds = 'username=' + username + '&password=' + password + '&'
+    else:
+        setcreds = ''
+
+    url = base_url + "/json.htm?" + setcreds + "type=command&param=switchlight&idx=" + str(idx) + "&switchcmd=Set%20Level&level=" + state
+    f = open(r"c:\tmp\debug.txt", "a")
+    f.write(url + "\n")
+    f.close()
+    requests.get(url=url)
+
+
+def get_list(optionsDict):
+    optionsList = []
+    for line in optionsDict:
+        optionsList.append(line['Name'])
+    optionsList.append("------End of list------")
```

Comment:
If this is shown in the UI it has to be localized

Reply:
fixed